example/crfutils.py

```diff
@@ -19,13 +21,17 @@ def apply_templates(X, templates):
     @type   template:   tuple of (str, int)
     @param  template:   The feature template.
     """
+    
+    x_range = list(range(len(X)))
+    x_range_set = set(x_range)
```

Comment:
I'm not sure if I understand the rationale behind this. The list cast effectively pre-allocates the list, which seems unnecessary - and the named variable declaration seems like an unnecessary change.